python-avd/pyavd/_eos_designs/structured_config/base/utils.py

```diff
@@ -121,20 +121,24 @@ def _get_vrf_and_source_interface(
         source_interface: str | None = None
         vrf = self.get_vrf(vrf_input, context=context)
         if set_source_interfaces:
-            # source_interface may be overridden below if given in the 'vrfs'
-            match vrf_input:
-                case None | "" | "use_default_mgmt_method_vrf":
-                    source_interface = self.shared_utils.default_mgmt_protocol_interface
-                case "use_mgmt_interface_vrf":
-                    source_interface = self.shared_utils.mgmt_interface
-                case "use_inband_mgmt_vrf":
-                    source_interface = self.shared_utils.inband_mgmt_interface
-
-        if vrf in vrfs and vrfs[vrf].source_interface:
-            source_interface = vrfs[vrf].source_interface
+            source_interface = self.get_source_interface(vrf_input, source_interface_override=vrfs[vrf].source_interface if vrf in vrfs else None)
 
         return (vrf, source_interface)
 
+    def get_source_interface(self: AvdStructuredConfigBaseProtocol, vrf_input: str | None, source_interface_override: str | None) -> str | None:
+        """Returns source interface for the given vrf, letting the given override take precedence."""
+        # source_interface may be overridden below if given in the 'vrfs'
```

Comment:
I am not sure this comment makes sense anymore
```suggestion
        # if an override is provided return it
```